c/tskit/tables.c

```diff
@@ -643,6 +643,40 @@ tsk_individual_table_truncate(tsk_individual_table_t *self, tsk_size_t num_rows)
     return ret;
 }
 
+int
+tsk_individual_table_extend(tsk_individual_table_t *self,
+    const tsk_individual_table_t *other, tsk_size_t num_rows,
+    const tsk_id_t *row_indexes, tsk_flags_t TSK_UNUSED(options))
+{
+    int ret = 0;
+    tsk_size_t j;
+    tsk_individual_t individual;
+
+    /* We know how much to expand the non-ragged columns, so do it ahead of time */
+    if (self->max_rows < self->num_rows + num_rows) {
+        ret = tsk_individual_table_expand_main_columns(self, num_rows);
+        if (ret != 0) {
+            goto out;
+        }
+    }
+    for (j = 0; j < num_rows; j++) {
+        ret = tsk_individual_table_get_row(
```

Comment:
Ah, really nice how this automatically does all our range checking for us! So if row_indexes is NULL and ``num_rows`` is bigger than the table, we'll get an out-of-bounds, exactly as we should. Love it.

Reply:
There's a small perf cost, that could be avoided in the NULL case as we could check the range and then copy the columns as a `memcpy` (apart from the offsets). Lets do that if we see the need, otherwise just added complexity at this point.

Reply:
yeah, we could special case this later if becomes an issue. For now, this is great.